From 295613ccc836f1fe3bf8f491cd3436c6403a9f0e Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Fri, 06 Aug 2021 09:35:28 +0300
Subject: [PATCH] [SW-50869] ppboot: make rst_src_dump as weak function

Gaudi1 and Greco has common bit definitions in PSOC_GLOBAL_CONF_RST_SRC
register for reset cause. But in Gaudi2 reset cause bit definitions are
modified and not backward compatible.

made rst_src_dump function weak so that it can be overwritten for Gaudi2
in later patches

Signed-off-by: Sailaja <sbandarupalli@habana.ai>
Change-Id: I398cc354811c7f4ed86ab6128a1e1653b6e7c5da
---

diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index c8f857c..5c74941 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -127,7 +127,7 @@
 
 }
 
-static void rst_src_dump(void)
+__weak void rst_src_dump(void)
 {
 	unsigned int rst_src = regmap_readl(mmPSOC_GLOBAL_CONF_RST_SRC);
 	char *cause;
